### ISSUE TITLE
When suggesting word by word reading, do not assume bash in a POSIX compliant script

### ISSUE THREAD

**Issue by oliv3r** [koalaman/shellcheck#1513]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2013
- My shellcheck version (`shellcheck --version` or "online"): online
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

for word in $(cat "/proc/cmdline"); do
    echo "${word}"
done

```

#### Here's what shellcheck currently says:
SC2013: To read lines rather than words, pipe/redirect to a 'while read' loop.

If you want to read word by word, you should still use a while read loop (e.g. with read -a to read words into an array).

#### Here's what I wanted or expected to see:
For one, shellcheck could detect I am trying to read word by word, as the word is passed into a switch case. Secondly, maybe by using 'word' as a variable name, this would indicate this? (though I am using cmd, which may indicate that), or the fact that /proc/cmdline is being read. But given that shellcheck cannot reliably detect this intention, the wiki is recommendations do not apply.

For one, read -a is not POSIX and secondly we can't read the words into an array, as that's not available. Thirdly 'for rare reasons you only care about numbers and not good practice' is only being offensive rather then constructive. Again, one may actually need to read words and cannot use arrays. So if there are other good methods to do this, the wiki should suggest those.

**Comment by xPMo** [comment]
This message is only an `info`-level error, indicating that the shells behavior may not be as you expect.
Anything at the `info` level or below can be correct in many circumstances, and it should be read as shellcheck reminding you of the implications.

(**Opinion**: adding a `#shellcheck disable=` line is a good way to document unorthodox behaviors in  your code. I use it often. Admittedly, in your case using the variable name `word` is currently doing that for you here.)

I do agree an edit to the wiki page to remove the unhelpful language is in order.  Care to suggest an edit?

**Comment by oliv3r** [comment]
I actually already am using the shellcheck disable; This ticket is mostly about whether shellcheck is able to detect such a thing. For example, we also have the special character _ which allows us to make note of things.
```sh
for _ in ${var}; do
```
will tell shellcheck that it is ok that the variable is unused (dummy var).

So again, the question really is, could shellcheck detect this usage, that the intention is to read a word, not the whole line?

If the answer is no; then the only solution of course is the shellcheck disable line and updating the wiki :)

**Comment by koalaman** [comment]
I updated the wiki, thanks!

Unfortunately I don't have any good suggestions for recognizing when this is intentional. Using the name `word` does not appear to be a convention in the way `_` is, and a sample from GitHub confirms what you say about the variables having diverse names. Maybe if `IFS` is explicitly set and `set -f` is used? It's somewhat tricky to deduce though. 

Without any better suggestions, I'll close this as a documentation fix
